### PR TITLE
Add login test

### PR DESCRIPTION
**Pull request by azizullahsaeidi**
Add Test For

1. Login
2. Forgot Password

### CODE REVIEW

**Comment by azizullahsaeidi** [comment]
> Hi @azizullahsaeidi, thank you for working on this. I ran tests locally and some tests failed. Wonder how the test workflow check passed against this branch.
> 
> ```
> $ php artisan test
> 
>    FAIL  Tests\Feature\Auth\ForgotPasswordTest
>   ✓ en user can view forgot password page
>   ⨯ en getting error if email does not exist
>   ✓ fa user can view forgot password page
>   ⨯ fa getting error if email does not exist
> 
>    FAIL  Tests\Feature\Auth\LoginTest
>   ✓ en guest can visit login page
>   ⨯ en user can login with valid credential
>   ⨯ en user cannot login with invalid password
>   ⨯ en user cannot login with invalid email
>   ✓ fa guest can visit login page
>   ⨯ fa user can login with valid credential
>   ⨯ fa user cannot login with invalid password
>   ⨯ fa user cannot login with invalid email
> 
>    PASS  Tests\Feature\ExampleTest
>   ✓ basic test
> 
>    PASS  Tests\Feature\Glossary\GlossaryListTest
>   ✓ user can visit glossary page
> 
>    PASS  Tests\Feature\HomePage\HomePageTest
>   ✓ user can visit english home page
>   ✓ user can visit farsi home page
>   ✓ user can visit pashto home page
>   ✓ user can visit uzbaki home page
>   ✓ user can visit munji home page
>   ✓ user can visit noorestani home page
>   ✓ user can visit sowji home page
>   ✓ user can visit shafnani home page
>   ✓ user can visit pashaiee home page
> 
>    PASS  Tests\Unit\ResourceTest
>   ✓ basic test
> 
>    PASS  Tests\Unit\UserTest
>   ✓ example
> 
>   ---
> 
>   • Tests\Feature\Auth\ForgotPasswordTest > en getting error if email does not exist
>   Session is missing expected key [errors].
>   Failed asserting that false is true.
> 
>   at tests/Feature/Auth/ForgotPasswordTest.php:38
>      34▕         User::factory()->create();
>      35▕ 
>      36▕         $response = $this->post('/en/password/email', ['email' => 'invalid@mail.com']);
>      37▕ 
>   ➜  38▕         $response->assertSessionHasErrors(['email' => "We can't find a user with that e-mail address."]);
>      39▕     }
>      40▕ 
>      41▕     /** @test */
>      42▕     public function fa_user_can_view_forgot_password_page()
> 
>   • Tests\Feature\Auth\ForgotPasswordTest > fa getting error if email does not exist
>   Session is missing expected key [errors].
>   Failed asserting that false is true.
> 
>   at tests/Feature/Auth/ForgotPasswordTest.php:64
>      60▕         User::factory()->create();
>      61▕ 
>      62▕         $response = $this->post('/fa/password/email', ['email' => 'invalid@mail.com']);
>      63▕ 
>   ➜  64▕         $response->assertSessionHasErrors(['email' => "کاربری با این ایمیل یافت نشد."]);
>      65▕     }
>      66▕ }
>      67▕ 
> 
>   • Tests\Feature\Auth\LoginTest > en user can login with valid credential
>   Expected response status code [201, 301, 302, 303, 307, 308] but received 419.
>   Failed asserting that false is true.
> 
>   at tests/Feature/Auth/LoginTest.php:44
>      40▕             'user-field' => 'user@email.com',
>      41▕             'password' => 'Pass@123',
>      42▕         ]);
>      43▕ 
>   ➜  44▕         $response->assertRedirect('home');
>      45▕         $this->assertAuthenticatedAs($user);
>      46▕         
>      47▕     }
>      48▕ 
> 
>   • Tests\Feature\Auth\LoginTest > en user cannot login with invalid password
>   Expected response status code [201, 301, 302, 303, 307, 308] but received 419.
>   Failed asserting that false is true.
> 
>   at tests/Feature/Auth/LoginTest.php:69
>      65▕             'user-field' => $user->email,
>      66▕             'password' => 'Invalid@123',
>      67▕         ]);
>      68▕         
>   ➜  69▕         $response->assertRedirect('/en/login');
>      70▕         $response->assertSessionHasErrors('email');
>      71▕         $response->assertSessionHasErrors();
>      72▕         $response->assertSessionHasErrors(['email' => "These credentials do not match our records."]);
>      73▕ 
> 
>   • Tests\Feature\Auth\LoginTest > en user cannot login with invalid email
>   Expected response status code [201, 301, 302, 303, 307, 308] but received 419.
>   Failed asserting that false is true.
> 
>   at tests/Feature/Auth/LoginTest.php:89
>      85▕             'user-field' => 'invalid@mail.com',
>      86▕             'password' => "Pass@123",
>      87▕         ]);
>      88▕ 
>   ➜  89▕         $response->assertRedirect('/en/login');
>      90▕         $response->assertSessionHasErrors('email');
>      91▕         $response->assertSessionHasErrors();
>      92▕         $response->assertSessionHasErrors(['email' => "These credentials do not match our records."]);
>      93▕ 
> 
>   • Tests\Feature\Auth\LoginTest > fa user can login with valid credential
>   Expected response status code [201, 301, 302, 303, 307, 308] but received 419.
>   Failed asserting that false is true.
> 
>   at tests/Feature/Auth/LoginTest.php:131
>     127▕             'user-field' => 'fa@email.com',
>     128▕             'password' => 'Pass@123',
>     129▕         ]);
>     130▕ 
>   ➜ 131▕         $response->assertRedirect('home');
>     132▕         $this->assertAuthenticatedAs($user);
>     133▕         
>     134▕     }
>     135▕ 
> 
>   • Tests\Feature\Auth\LoginTest > fa user cannot login with invalid password
>   Expected response status code [201, 301, 302, 303, 307, 308] but received 419.
>   Failed asserting that false is true.
> 
>   at tests/Feature/Auth/LoginTest.php:156
>     152▕             'user-field' => $user->email,
>     153▕             'password' => 'Invalid@123',
>     154▕         ]);
>     155▕         
>   ➜ 156▕         $response->assertRedirect('/fa/login');
>     157▕         $response->assertSessionHasErrors('email');
>     158▕         $response->assertSessionHasErrors();
>     159▕         $response->assertSessionHasErrors(['email' => "اطلاعات وارد شده غ��ط میباشد."]);
>     160▕         
> 
>   • Tests\Feature\Auth\LoginTest > fa user cannot login with invalid email
>   Expected response status code [201, 301, 302, 303, 307, 308] but received 419.
>   Failed asserting that false is true.
> 
>   at tests/Feature/Auth/LoginTest.php:176
>     172▕             'user-field' => 'invalid@mail.com',
>     173▕             'password' => "Pass@123",
>     174▕         ]);
>     175▕ 
>   ➜ 176▕         $response->assertRedirect('/fa/login');
>     177▕         $response->assertSessionHasErrors('email');
>     178▕         $response->assertSessionHasErrors();
>     179▕         $response->assertSessionHasErrors(['email' => "اطلاعات وارد شده غلط میباشد."]);
>     180▕         
> 
> 
>   Tests:  8 failed, 17 passed
>   Time:   26.38s
> ```

Dear @uyscuti-wiki , I have tested this PR on local and got no error.
image

**Comment by uyscuti-wiki** [comment]
@azizullahsaeidi, are you certain you are running your branch is up-to-date with master? I can see from your logs that tests from `Tests\Feature\Auth\ForgotPasswordTest` and `Tests\Feature\Auth\LoginTest` are not executed.

**Comment by azizullahsaeidi** [comment]
> @azizullahsaeidi, are you certain you are running your branch is up-to-date with master? I can see from your logs that tests from `Tests\Feature\Auth\ForgotPasswordTest` and `Tests\Feature\Auth\LoginTest` are not executed.

Got you @uyscuti-wiki , Thank you

**Comment by azizullahsaeidi** [comment]
> > @azizullahsaeidi, are you certain you are running your branch is up-to-date with master? I can see from your logs that tests from `Tests\Feature\Auth\ForgotPasswordTest` and `Tests\Feature\Auth\LoginTest` are not executed.
> 
> Got you @uyscuti-wiki , Thank you

Dear @uyscuti-wiki, issue has been solved, please check it again. Thanks

File: tests/Feature/Auth/ForgotPasswordTest.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    /** @test */
+    public function en_getting_error_if_email_does_not_exist()
+    {
+        $this->refreshApplicationWithLocale('en');
+
+        User::factory()->create();
+
+        $response = $this->post('/en/password/email', ['email' => 'invalid@mail.com']);
+
+        $response->assertSessionHasErrors(['email' => "We can't find a user with that e-mail address."]);
```
Comment (uyscuti-wiki) [inline tests/Feature/Auth/ForgotPasswordTest.php:38]: While acceptable for now. This cannot be. See [DDL-117](https://ddlibrary.atlassian.net/jira/software/c/projects/DDL/boards/3?selectedIssue=DDL-117).

File: tests/Feature/Auth/LoginTest.php [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        $response->assertRedirect('/en/login');
+        $response->assertSessionHasErrors('email');
+        $response->assertSessionHasErrors();
+        $response->assertSessionHasErrors(['email' => "These credentials do not match our records."]);
+
+        $this->assertFalse(session()->hasOldInput('password'));
+        $this->assertGuest();
+    }
+
+
```
Comment (uyscuti-wiki) [inline tests/Feature/Auth/LoginTest.php:113]: We should add a test to ensure the throttling trait is working as intended, specifically the `hasTooManyLoginAttempts()` function.

**Review by uyscuti-wiki — CHANGES_REQUESTED** [review]
Hi @azizullahsaeidi, thank you for fixing the failing tests. I have one more request for you. Should be good to merge after that.

File: tests/Feature/Auth/LoginTest.php (hunk 2 above)
Comment (azizullahsaeidi) [reply tests/Feature/Auth/LoginTest.php]: Thank you @uyscuti-wiki , done.

File: tests/Feature/Auth/ForgotPasswordTest.php (hunk 1 above)
Comment (azizullahsaeidi) [reply tests/Feature/Auth/ForgotPasswordTest.php]: Message has been changed.

**Review by uyscuti-wiki — APPROVED** [review]
Looks great. Thank you!

**Pull request merged**